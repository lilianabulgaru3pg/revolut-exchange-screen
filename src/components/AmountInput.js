import React from 'react';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const TextField = styled.input`
  text-align: right;
  color: #1e90ff;
  caret-color: #1e90ff;
  border: none;
  font-size: inherit;
  font-family: inherit;
  font-weight: inherit;
  line-height: inherit;
  letter-spacing: inherit;
  background-color: transparent;
  &::-webkit-outer-spin-button,
  &::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
  &::-webkit-input-placeholder {
    color: #b0c4de;
  }
  &:focus {
    outline: none;
  }
`;

const AmountInput = ({
  classes,
  autoFocus,
  inputVal,
  onInputChange,
}) => {
  return (
    <Typography component="div">
      <TextField
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        pattern="[0-9.]*"
        type="number"
        onChange={({ target: { value } }) => onInputChange(value)}
        value={inputVal > 0 ? inputVal : ''}
        autoFocus={autoFocus}
        className={classes}
        placeholder="0"
      />
    </Typography>
  );
};

AmountInput.propTypes = {
  onInputChange: PropTypes.func.isRequired,
  classes: PropTypes.string,
  autoFocus: PropTypes.bool,
  inputVal: PropTypes.number.isRequired,
};

AmountInput.defaultProps = {
  classes: '',
  autoFocus: false,
};

export default AmountInput;
