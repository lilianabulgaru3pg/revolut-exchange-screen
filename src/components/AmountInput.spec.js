import React from 'react';
import renderer from 'react-test-renderer';
import AmountInput from './AmountInput';

describe('AmountInput Component', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <AmountInput
        autoFocus
        inputVal={120}
        onInputChange={jest.fn()}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
