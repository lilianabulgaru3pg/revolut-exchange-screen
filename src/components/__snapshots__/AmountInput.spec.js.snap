// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountInput Component snapshot renders 1`] = `
<div
  className="MuiTypography-root MuiTypography-body1"
>
  <input
    autoCapitalize="off"
    autoComplete="off"
    autoCorrect="off"
    autoFocus={true}
    className="sc-AykKC exZWIY"
    onChange={[Function]}
    pattern="[0-9.]*"
    placeholder="0"
    type="number"
    value={120}
  />
</div>
`;
